rest-server/src/controllers/user.js

```diff
@@ -32,84 +32,74 @@ const update = (req, res) => {
       (typeof admin === 'undefined' || !admin) && modify) {
     userModel.update(username, password, admin, modify, (err, state) => {
       if (err || !state) {
-        logger.warn('update user %s failed', username);
-        return res.status(500).json({
-          error: 'UpdateFailed',
-          message: 'update user failed',
-        });
+        const error = err || new Error('update user failed');
+        error.status = httpStatus.INTERNAL_SERVER_ERROR;
+        next(error);
       } else {
-        return res.status(201).json({
+        return res.status(httpStatus.CREATED).json({
           message: 'update successfully',
         });
       }
     });
   } else {
-    return res.status(401).json({
-      error: 'NotAuthorized',
-      message: 'not authorized',
-    });
+    const error = new Error('not authorized');
+    error.status = httpStatus.UNAUTHORIZED;
+    next(error);
   }
 };
 
 /**
  * Remove a user.
  */
-const remove = (req, res) => {
+const remove = (req, res, next) => {
   const username = req.body.username;
   if (req.user.admin) {
     userModel.remove(username, (err, state) => {
       if (err || !state) {
-        logger.warn('remove user %s failed', username);
-        return res.status(500).json({
-          error: 'RemoveFailed',
-          message: 'remove failed',
-        });
+        const error = err || new Error('remove failed');
+        error.status = httpStatus.INTERNAL_SERVER_ERROR;
+        next(error);
       } else {
```

Comment:
We should provide "error.message" always in our whole project to give users a clear information includes What and Why, better with How to hint user solve the problem.